mod/NomaiTextQoL.cs

```diff
@@ -1,42 +1,146 @@
+                    if (__instance._textLines.Length == 1)
+                    {
+                        CheckHintData hintData = __instance._textLines[0].gameObject.GetAddComponent<CheckHintData>();
+
+                        // DatabaseID is the ship log name
+                        string log = nomaiTextData.DatabaseID;
+
+                        // Check for Location Triggers
+                        if (LocationTriggers.logFactToDefaultLocation.ContainsKey(log))
+                        {
+                            hintData.DetermineImportance(LocationTriggers.logFactToDefaultLocation[log]);
+                        }
+
+                        // Check for Logsanity checks
+                        bool isALog = Enum.TryParse<Location>("SLF__" + nomaiTextData.DatabaseID, out Location loc);
+                        if (isALog && APRandomizer.SlotData.ContainsKey("logsanity") && (long)APRandomizer.SlotData["logsanity"] != 0)
+                        {
+                            hintData.DetermineImportance(loc);
+                        }
+                        continue;
+                    }
+
+                    for (int j = 0; j < nomaiTextData.ConditionBlock.Length; j++)
+                    {
+                        for (int k = 0; k < nomaiTextData.ConditionBlock[j].Length; k++)
+                        {
+                            int key = nomaiTextData.ConditionBlock[j][k];
+                            // if this succeeds we've found an arc that matches the conditions, and thus we can find which log is attached
+                            // I don't understand it either, blame Mobius making this system far more complicated than it needed to be
+                            if (__instance._dictNomaiTextData.ContainsKey(key))
+                            {
+                                var textLine = __instance._textLines.First(x => x.GetEntryID() == key);
+
+                                CheckHintData hintData = textLine.gameObject.GetAddComponent<CheckHintData>();
+
+                                // DatabaseID is the ship log name
+                                string log = nomaiTextData.DatabaseID;
+
+                                // Check for Location Triggers
+                                if (LocationTriggers.logFactToDefaultLocation.ContainsKey(log))
+                                {
+                                    hintData.DetermineImportance(LocationTriggers.logFactToDefaultLocation[log]);
+                                }
+
+                                // Check for Logsanity checks
+                                bool isALog = Enum.TryParse<Location>("SLF__" + nomaiTextData.DatabaseID, out Location loc);
+                                if (isALog && APRandomizer.SlotData.ContainsKey("logsanity") && (long)APRandomizer.SlotData["logsanity"] != 0)
+                                {
+                                    hintData.DetermineImportance(loc);
+                                }
```

Comment:
Looks like this is almost identical to the single-line loop above, except for the text line fetching. Could we add a helper method like `DetermineImportance/ColorFor(textLine)`?

Reply:
Sure, but I don't see how this improves the code at all. It's two lines of code.

Reply:
Actually, no, it's used differently in each spot. Not worth it.